crates/keyshare/src/threshold_keyshare.rs

```diff
@@ -767,13 +771,63 @@ impl Handler<EnclaveEvent> for ThresholdKeyshare {
 impl Handler<CiphernodeSelected> for ThresholdKeyshare {
     type Result = ();
     fn handle(&mut self, msg: CiphernodeSelected, ctx: &mut Self::Context) -> Self::Result {
+        // Store selection for potential committee finalization
+        self.pending_selections
+            .insert(msg.e3_id.clone(), msg.clone());
+
+        // If CommitteeFinalized arrives later, it will verify committee membership
```

Comment:
`handle_ciphernode_selected` will start the key generation process which is very expensive - are we sure we want to do this before the committee has been finalized and we know we are part of the committee? Maybe this makes sense but it seems like it might be premature optimization.